crates/wasmtime/src/runtime/vm.rs

```diff
@@ -336,6 +336,216 @@ impl ModuleRuntimeInfo {
+        }
+    }
+
+    /// Creates a new `HostAlignedByteCount` from a `u64`, rounding up to the nearest page.
+    ///
+    /// Returns an error if the `u64` overflows `usize`, or if `bytes +
+    /// page_size - 1` overflows.
+    pub fn new_rounded_up_u64(bytes: u64) -> Result<Self, ByteCountOutOfBounds> {
+        let bytes = bytes
+            .try_into()
+            .map_err(|_| ByteCountOutOfBounds(ByteCountOutOfBoundsKind::U64(bytes)))?;
+        Self::new_rounded_up(bytes)
+    }
+
+    /// Returns the host page size.
+    pub fn host_page_size() -> HostAlignedByteCount {
+        // The host page size is always a multiple of itself.
+        HostAlignedByteCount(host_page_size())
+    }
+
+    /// Returns true if the page count is zero.
+    #[inline]
+    pub fn is_zero(self) -> bool {
+        self == Self::ZERO
+    }
+
+    /// Returns the number of bytes as a `usize`.
+    #[inline]
+    pub fn byte_count(self) -> usize {
+        self.0
+    }
+
+    /// Add two aligned byte counts together.
+    ///
+    /// Returns an error if the result overflows.
+    pub fn checked_add(self, bytes: HostAlignedByteCount) -> Result<Self, ByteCountOutOfBounds> {
+        // aligned + aligned = aligned
+        self.0
+            .checked_add(bytes.0)
+            .map(Self)
+            .ok_or(ByteCountOutOfBounds(ByteCountOutOfBoundsKind::Add(
+                self.0, bytes.0,
+            )))
+    }
+
+    /// Subtract two aligned byte counts.
+    ///
+    /// Returns an error if the result underflows.
+    pub fn checked_sub(self, bytes: HostAlignedByteCount) -> Result<Self, ByteCountOutOfBounds> {
+        // aligned - aligned = aligned
+        self.0
+            .checked_sub(bytes.0)
+            .map(Self)
+            .ok_or(ByteCountOutOfBounds(ByteCountOutOfBoundsKind::Sub(
+                self.0, bytes.0,
+            )))
+    }
+
+    /// Multiply an aligned byte count by a scalar value.
+    ///
+    /// Returns an error if the result overflows.
+    pub fn checked_mul(self, n: usize) -> Result<Self, ByteCountOutOfBounds> {
+        // aligned * scalar = aligned
+        self.0
+            .checked_mul(n)
+            .map(Self)
+            .ok_or(ByteCountOutOfBounds(ByteCountOutOfBoundsKind::Mul(
+                self.0, n,
+            )))
+    }
+
+    /// Unchecked multiplication by a scalar value.
+    ///
+    /// ## Safety
+    ///
+    /// The result must not overflow.
+    #[inline]
+    pub unsafe fn unchecked_mul(self, n: usize) -> Self {
+        Self(self.0 * n)
+    }
+}
+
+impl PartialEq<usize> for HostAlignedByteCount {
+    #[inline]
+    fn eq(&self, other: &usize) -> bool {
+        self.0 == *other
+    }
+}
+
+impl PartialEq<HostAlignedByteCount> for usize {
+    #[inline]
+    fn eq(&self, other: &HostAlignedByteCount) -> bool {
+        *self == other.0
+    }
+}
+
+impl fmt::Display for HostAlignedByteCount {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.0.fmt(f)
```

Comment:
Should this print the size in hex or as a decimal?

Reply:
Let's go with hex perhaps to clearly show that the low bits are all zero